### PR TITLE
PUBDEV-8245 Prevent R connect from starting h2o locally

### PR DESCRIPTION
**Pull request by g-eoj**
This fixes the error `Can only start H2O launcher if IP address is localhost.` when the user trys to connect to a non-localhost cluster using an `ip` argument and the cluster is not found.

Fixes https://h2oai.atlassian.net/browse/PUBDEV-8245

### CODE REVIEW

File: h2o-r/h2o-package/R/connection.R [hunk 1]
```diff
@@ -335,7 +334,7 @@ h2o.connect <- function(ip = "localhost", port = 54321, strict_version_check = T
    do.call(h2o.init, c(startH2O=FALSE, config))
  } else {
    # Pass arguments directly
-   h2o.init(ip=ip, port=port, strict_version_check=strict_version_check,
+   h2o.init(ip=ip, port=port, startH2O=FALSE, strict_version_check=strict_version_check,
```
Comment (sebhrusen) [inline h2o-r/h2o-package/R/connection.R:337]: it's interesting this reversal between R and Py implementations:
- in Py, `h2o.init()` calls `h2o.connect()`
- in R, `h2o.connect()` calls `h2o.init()`…, hence the issue at the origin of this ticket.

this looks like the right way to fix it in this context.

File: h2o-r/tests/testdir_misc/runit_connect.R [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            "kvstore.R", "frame.R", "targetencoder.R", "astfun.R","automl.R", "import.R", "parse.R", "export.R", "models.R", "edicts.R",
+            "coxph.R", "coxphutils.R", "gbm.R", "glm.R", "gam.R", "glrm.R", "kmeans.R", "deeplearning.R", "randomforest.R", "generic.R",
+            "naivebayes.R", "pca.R", "svd.R", "locate.R", "grid.R", "word2vec.R", "w2vutils.R", "stackedensemble.R", "rulefit.R",
+            "predict.R", "xgboost.R", "isolationforest.R", "psvm.R", "segment.R", "tf-idf.R", "explain.R", "permutation_varimp.R", "extendedisolationforest.R")
+src_path <- paste(h2oRDir,"h2o-package","R",sep=.Platform$file.sep)
+invisible(lapply(to_src,function(x){source(paste(src_path, x, sep = .Platform$file.sep))}))
+
+source("../runitUtils/utilsR.R")
+default.packages()
+Log.info("Loaded default packages. Additional required packages must be loaded explicitly.")
```
Comment (sebhrusen) [inline h2o-r/tests/testdir_misc/runit_connect.R:16]:
```suggestion
source("../../scripts/h2o-r-test-setup.R")
```
this should be enough.
Did you have any problem with the approach used in other tests?

File: h2o-r/tests/testdir_misc/runit_connect.R [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }, error = function(x) x)
+  
+  expect_false(is.null(e))
+  print(e)
+  err_message <- e[[1]]
+  expect_true("Cannot connect to H2O server. Please check that H2O is running at http://localhost:54321/" == err_message)
+}
+
+doTest("Error thrown when h2o.connect() is called on non-existent cluster (local cluster)",
+       h2o_connect_test)
```
Comment (sebhrusen) [inline h2o-r/tests/testdir_misc/runit_connect.R:31]:
```suggestion
doTest("Test that h2o.connect() doesn't start H2O server locally",
       h2o_connect_test)
```
The string arg is supposed to describe the test and/or its expected behaviour, it's not clear if your message describes what happenes on success or on failure.

**Review by sebhrusen — COMMENTED** [review]
Thanks @g-eoj 
Added some suggestions for the tests.

File: h2o-r/tests/testdir_misc/runit_connect.R (hunk 2 above)
Comment (g-eoj) [reply h2o-r/tests/testdir_misc/runit_connect.R:16]: Yes, `source("../../scripts/h2o-r-test-setup.R")` seems like it expects h2o to be running in the background.

File: h2o-r/tests/testdir_misc/runit_connect.R (hunk 3 above)
Comment (g-eoj) [reply h2o-r/tests/testdir_misc/runit_connect.R:31]: Then `"Test that h2o.connect() throws error when connecting to a non-existent cluster"`?

**Review by ledell — APPROVED** [review]
This works, thanks.  
```
> h2o.connect()
Reading in config file: ./../.h2oconfig
Error in h2o.init(ip = ip, port = port, startH2O = FALSE, strict_version_check = strict_version_check,  :
  Cannot connect to H2O server. Please check that H2O is running at http://localhost:54321/
```

@sebhrusen Are the tests working as you expected?

**Pull request merged**